### PR TITLE
Integrate the api dependency from Infinispan 14  (#ISPN-14268)

### PR DESCRIPTION
**Pull request by karesti**
A new dependency added to allow to integrate new features

* Indexing annotations (documentation updated)
* Infinispan 14.0.2.Final minor update

### CODE REVIEW

File: docs/src/main/asciidoc/infinispan-client.adoc [hunk 1]
```diff
@@ -109,15 +109,42 @@ quarkus.infinispan-client.cache.magazine.configuration=<distributed-cache><encod
 If both `configuration-uri` and `configuration` are configured for the same cache with the same Quarkus profile,
 `configuration-uri` gets preference over `configuration`.
 
+[TIP]
+====
+Cache configuration can be provided in XML, JSON or YAML. Use the Infinispan Console and the cache configuration Wizard
+to learn more about Infinispan Caches and create guided configurations.
```
Comment (geoand) [inline docs/src/main/asciidoc/infinispan-client.adoc:115]: It would be awesome in the future if this where possible using the DevUI

**Review by geoand — APPROVED** [review]

File: docs/src/main/asciidoc/infinispan-client.adoc (hunk 1 above)
Comment (karesti) [reply docs/src/main/asciidoc/infinispan-client.adoc:115]: From the Dev UI we have a link to the console, I don't know if it's possible to bring the console framed into the dev ui

Comment (geoand) [reply docs/src/main/asciidoc/infinispan-client.adoc:115]: It should be. @phillip-kruger?

Comment (phillip-kruger) [reply docs/src/main/asciidoc/infinispan-client.adoc:115]: Yes definitely possible.  In the (soon to be) new Dev UI you can use the external link and that should happen automatically.

Comment (geoand) [reply docs/src/main/asciidoc/infinispan-client.adoc:115]: That's awesome!

**Pull request merged**